### PR TITLE
Ltac2 ml_intern does not use the "self" recursive intern function

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: plugins/ltac2/tac2env.mli [hunk 1]
```diff
@@ -157,7 +157,7 @@ type environment = {
 }
 
 type ('a, 'b) ml_object = {
-  ml_intern : 'r. (raw_tacexpr, glb_tacexpr, 'r) intern_fun -> ('a, 'b or_glb_tacexpr, 'r) intern_fun;
+  ml_intern : 'r. ('a, 'b or_glb_tacexpr, 'r) intern_fun;
```
[the remark this answers is not in the document]
Comment (ppedrot) [reply plugins/ltac2/tac2env.mli:160]: Indeed there may have been some motivation behind this code that was never used in practice.

**Review by ppedrot — APPROVED** [review]
I'm fine with the removal, since we're not really relying on that currently. I'd really like to understand what feature I had in mind with this though.

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**